### ISSUE TITLE
Viewer doesn't properly work for MathJax parts in Markdown cells

### ISSUE THREAD

**Issue by henkelma** [tillahoffmann/obsidian-jupyter#36]
MathJax parts are only properly rendered if the window is in focus while Obsidian is started. In all other cases the MathJax source is visible instead to the rendered equations.

Seems to be caused by:

```js
// https://stackoverflow.com/a/47614491/1150961.
function setInnerHTML(elm, html) {
    elm.innerHTML = html;
    Array.from(elm.querySelectorAll("script")).forEach(oldScript => {
        const newScript = document.createElement("script");
        Array.from(oldScript.attributes)
            .forEach(attr => newScript.setAttribute(attr.name, attr.value));
        newScript.appendChild(document.createTextNode(oldScript.innerHTML));
        oldScript.parentNode.replaceChild(newScript, oldScript);
    });
}
```

I replaced this with

```js
function setInnerHTML(elm, text) {
    elm.innerHTML = text;
    var scripts = Array.prototype.slice.call(elm.getElementsByTagName("script"));
    for (var i = 0; i < scripts.length; i++) {
        if (scripts[i].src != "") {
            var tag = document.createElement("script");
            tag.src = scripts[i].src;
            document.getElementsByTagName("head")[0].appendChild(tag);
        }
        else {
            eval(scripts[i].innerHTML);
        }
    }
}
```

This leads to correctly rendered output in all cases. But during the first rendering the following exception is being raised:
```
Uncaught ReferenceError: init_mathjax is not defined
    at eval (eval at insertAndExecute (eval at <anonymous>```

**Comment by henkelma** [comment]
Forget the alternative code, it creates more problems than it actually solves.
Being new to the js/html world it is tough to judge everything in detail. But from what I guess it would probably be much cleaner to embed the whole Jupyter html into an iframe.
Need to experiment further an think about it...

**Comment by henkelma** [comment]
There seems to be something fishy with MathJax in v0.13.23 in general. Math in regular markdown files is also not rendered properly after switching back and forth a few times.

**Comment by henkelma** [comment]
There is an incompatibility with the used MathJax version between Obsidian and Jupyter. Obsidian is using v3 and Jupyter is using v2. For now that means, that the generated html from an ipynb file can only "live" in an iFrame. Otherwise there will be problems with MathJax.

**Comment by aidan-gibson** [comment]
#37

**Comment by tillahoffmann** [comment]
This is hopefully fixed by #40.
